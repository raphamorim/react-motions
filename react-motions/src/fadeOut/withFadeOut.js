import React from 'react'
import styles from './styles'

function withFadeOut(Component) {
  const style = {
    WebkitAnimationName: "fadeOut",
    animationName: "fadeOut",
    WebkitAnimationDuration: "1s",
    animationDuration: "1s",
    WebkitAnimationFillMode: "both",
    animationFillMode: "both"
  }

  return (
    <div style={style}>
      {Component}
    </div>
  )
}

export default withFadeOut
