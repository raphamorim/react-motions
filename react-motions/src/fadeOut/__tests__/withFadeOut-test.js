import React from 'react';
import withFadeOut from '../withFadeOut';

describe('withFadeOut', () => {
  it('should render', () => {
    const wrapper = shallow(withFadeOut(<div>FadeOut</div>));

    expect(wrapper).toMatchSnapshot();
  });
});
