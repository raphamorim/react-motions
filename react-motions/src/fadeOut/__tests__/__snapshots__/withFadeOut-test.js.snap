// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withFadeOut should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    style={
      Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "fadeOut",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "fadeOut",
      }
    }
  >
    <div>
      FadeOut
    </div>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        FadeOut
      </div>,
      "style": Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "fadeOut",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "fadeOut",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "FadeOut",
      },
      "ref": null,
      "rendered": "FadeOut",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          FadeOut
        </div>,
        "style": Object {
          "WebkitAnimationDuration": "1s",
          "WebkitAnimationFillMode": "both",
          "WebkitAnimationName": "fadeOut",
          "animationDuration": "1s",
          "animationFillMode": "both",
          "animationName": "fadeOut",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FadeOut",
        },
        "ref": null,
        "rendered": "FadeOut",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
